doc/stages/filters.covariancefeatures.rst

```diff
@@ -24,22 +43,49 @@ Example
       },
       {
           "type":"writers.bpf",
-          "filename":"output.las",
+          "filename":"output.bpf",
           "output_dims":"X,Y,Z,Linearity,Planarity,Scattering,Verticality"
       }
   ]
 
+Example #2
+-------------------------------------------------------------------------------
+
+.. code-block:: json
+
+  [
+      "input.las",
+      {
+          "type":"filters.optimalneighborhood"
+      },
+      {
+          "type":"filters.covariancefeatures",
+          "knn":8,
+          "threads": 2,
+          "optimized":true,
+          "features": "Linearity,Omnivariance,Density"
+      },
+      {
+          "type":"writers.las",
+          "minor_version":4,
+          "extra_dims":"all",
+          "forward":"all",
+          "filename":"output.las"
+      }
+  ]
+
 Options
 -------------------------------------------------------------------------------
 
 knn
-  The number of k nearest neighbors used for calculating the covariance matrix. [Default: 10]
+  The number of k nearest neighbors used for calculating the covariance matrix.
+  [Default: 10]
 
 threads
   The number of threads used for computing the feature descriptors. [Default: 1]
 
 feature_set
```

Comment:
I wouldn't. The original thinking was that there were particular groupings of said features that were desirable or in line with a particular approach. Granted we only have the one, but this gives a preset option that avoids needing to manually populate all desired features.

Reply:
I really don't understand.

Reply:
I missed the actual lines of code you were commenting on (thought it was `feature_set`). Why would you remove `knn`? Because of the `OptimalKNN` business we've introduced? It will certainly be superceded in that case, but `OptimalKNN` is not required, so you need something to fallback on.

Reply:
No. It was feature set. I don't get why you have one option but two sets of behavior. Your choice is `Dimensionality` or ... ?

Reply:
>Your choice is `Dimensionality` or ... ?

`Dimensionality` via `feature_set`, or explicitly listing individual features via `features`.

The default (and only option) previously had been `feature_set="Dimensionality"`. This PR changes that such that the default behavior is `features="all"`, but the user still has the option to 1) override with `feature_set="Dimensionality"` or 2) list individual `features`.

Reply:
Latest commit (https://github.com/PDAL/PDAL/pull/3180/commits/814978c77b5ab14b63d4ecf2e14c0e6ea28981e5) takes all feature related options (`feature_set` presets and individual `features`) via the `feature_sets` arg. Default reverted back to `Dimensionality`. So, now you could specify `feature_set="Dimensionality,Omnivariance"` to get `Linearity`, `Planarity`, `Scattering`, `Verticality` and `Omnivariance`.